### PR TITLE
Create smooth.thief

### PR DESCRIPTION
**Pull request by trnnick**
Wrapper to use smooth with thief

### CODE REVIEW

File: R/smooth.thief [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+         "ces" = {
+           fit <- auto.ces(data,silent=TRUE,...)
+         },
+         "ges" = {stop("No automatic GES model fitting yet")})
+  
+  fc <- forecast(fit,h=h)
+  if (type == "es" | type == "ces"){
+    fc$x <- fc$actuals
+  }
+  fc$lower <- fc$upper <- fc$level <- fc$forecast <- fc$intervals <- fc$residuals <- fc$method <- fc$xname <- fc$actuals <- NULL
```
Comment (config-i1) [inline R/smooth.thief]: This line looks strange. Is this function supposed to return NULL for all those values?

File: R/smooth.thief [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+         "ssarima" = {
+           fit <- auto.ssarima(data,silent=TRUE,...)
+         },
+         "ces" = {
+           fit <- auto.ces(data,silent=TRUE,...)
+         },
+         "ges" = {stop("No automatic GES model fitting yet")})
+  
+  fc <- forecast(fit,h=h)
+  if (type == "es" | type == "ces"){
```
Comment (config-i1) [inline R/smooth.thief]: Lines 54 - 56 are redundant. forecast.smooth() now also returns $x.

**Comment by config-i1** [comment]
Overall I don't mind merging it, but why is it here and not in THieF? Does it mean that smooth now should also depend on THieF? Doesn't make much sense for my taste. And why the name "smooth.thief"? This implies that there is a class "thief" and a method "smooth" for it...

**Comment by trnnick** [comment]
- The rest I remove (with NULL), as some are used in a different way by thief, and if they are provided and spew out an error. 
- I would not suggest a adding thief as a dependency (or the other way around), I kind of prefer keeping packages slim. I may be mistaken, but I think since the example does not run, it will not check for the dependency. However, I should probably add a check at the beginning of the function that checks that and gives an error if the package is not in memory.
- Name:fair enough. I just found smooth.thief funny, but whatever name you feel is appropriate!

**Comment by config-i1** [comment]
* It seems to me that this function fits better THieF package with "Suggests smooth" and a check if the package is loaded. Otherwise it sort of ruins logic: if you want to use lm() for similar purpose, you won't write a separate function for that in stats package, you would do that in THieF. And yes, it won't check the dependencies, if you don't run the example but that's a "hack" :).
* I like the name as well, but it should be consistent with other names in the package.

Sorry for being a downer, but I'm not sure that it is needed in smooth. Maybe I'm wrong, but I'm not convinced otherwise yet :).

**Comment by trnnick** [comment]
The way I see it, it is the opposite way, as thief already provides a generic interface for functions to connect with it. The other argument is that currently forecast and smooth provide some similar model families, so having that there would cause confusion, while people who use smooth, I assume, know the difference between this and forecast. But at the end of the day you call the shots! I will include it for now to TStools to let people quickly interface the two.

**Comment by config-i1** [comment]
Let me investigate Thief a bit, so I have a better understanding of what's happening. Then I'll make a desicion ;)

**Comment by config-i1** [comment]
I understand now how THieF works, but I still don't think that this function fits here. I'd propose you to change THieF function or introduce thief.smooth there.

**Pull request closed without merging**